packages/dds/merge-tree/src/endOfTreeSegment.ts

```diff
@@ -24,18 +25,18 @@ import { depthFirstNodeWalk, NodeAction } from "./mergeTreeNodeWalk";
  */
 export class EndOfTreeSegment implements ISegment, IRemovalInfo {
     type: string = "EndOfTreeSegment";
-    private readonly root: IMergeBlock;
+    private readonly mergeTree: MergeTree;
     constructor(segmentOrNode: IMergeNode) {
-        let maybeRoot: IMergeBlock | undefined = segmentOrNode.isLeaf()
-            ? segmentOrNode.parent
-            : segmentOrNode;
-        while (maybeRoot?.parent !== undefined) {
+        let maybeRoot: IRootMergeBlock | undefined = segmentOrNode.isLeaf()
+        ? segmentOrNode.parent
+        : segmentOrNode;
+        while(maybeRoot?.parent !== undefined) {
             maybeRoot = maybeRoot.parent;
         }
-        if (maybeRoot === undefined) {
+        if (maybeRoot === undefined || maybeRoot.mergeTree === undefined) {
```

Comment:
this can be simplified with `?` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining
```suggestion
        if (maybeRoot?.mergeTree === undefined) {
```